### ISSUE TITLE
Potential loopholes in form completion logic

### ISSUE THREAD

**Issue by chrispmad** [bcgov/invasivesBC-mussels-iOS#252]
**As a** *Data Analyst*, **I would greatly appreciate** *a rework/revisit of form completion logic* **So That** *it is impossible for an inspection record to be submitted if it is lacking information in fields flagged as required values.*

**Additional Context**
- Martina Beck and I assessed data gaps in key data fields (e.g. Destination Waterbody 1 Name, Closest City, Other Details, etc., and similar fields for Previous Waterbody 1) and found there to be a significant proportion of records (something like 10 to 30%) that were somehow submitted despite not having any information in any of these fields. My understanding is that there are multiple levels of logic, requiring info to be input to at least one of those fields, so Martina and I are not sure how we are finding these records that are missing destination/previous waterbody data. This seems like a tricky and open-ended issue, so if I can help in any way, please let me know!

- update status field to indicated complete/incomplete with a red dot for incomplete
- status field recorded in DB
- end shift workflow to include a prompt around incomplete inspections
- simplifying the deletion of incomplete records from the shift view
- *a user should not be able to delete a complete record, but they can edit*

**Acceptance Criteria**
- [ ] Given current date gaps, When an inspection officer attempts to submit an inspection record but all of these key fields are still NA/NULL/blank, Then they should not be able to submit the record.

**Comment by jakemorr** [comment]
Chris will attempt to reproduce the bug and proovide steps on how to cause it.

**Comment by chrispmad** [comment]
I managed to reproduce a loophole! I started a test shift, clicked the 'Add Inspection' button and filled out some of the mandatory fields. I did not select a destination nor a previous waterbody (nor did I click any of the toggles that switch the required info to be the nearest city). I then clicked 'back' to see the shift page. I then simply submitted the shift and, though the inspection was just a draft, it was submitted to the InvasivesBC db and is now visible on metabase (as a test). Wa ha!

**Comment by chrispmad** [comment]
I managed to reproduce a loophole! I started a test shift, clicked the 'Add Inspection' button and filled out some of the mandatory fields. I did not select a destination nor a previous waterbody (nor did I click any of the toggles that switch the required info to be the nearest city). I then clicked 'back' to see the shift page. I then simply submitted the shift and, though the inspection was just a draft, it was submitted to the InvasivesBC db and is now visible on metabase (as a test). Wa ha!
